### DISCUSSION TITLE
rabbitmq-diagnostic

### DISCUSSION THREAD

**Discussion by joaquinjsb** [rabbitmq/rabbitmq-server#5112] [Other] [answered]
Hello guys,
the tool rabbitmq-diagnostic fails on ARM like cpu architectures, running rabbitmq-diagnostics -q ping for example, returns a segmentation fault, creating problems with liveness/probeness on containers, therefore there's no reliable way of checking the status, I wouldn't exclude that the problem affects other tools. 

to replicate you just need to run on shell any of the rabbitmq-diagnostic functions on the shell on an operating system with arm.

**Comment by michaelklishin** [comment, marked as answer]
RabbitMQ and its CLI tools do not contain any architecture-specific code. They run wherever Erlang runs, so you need an image that contains Erlang built on and for ARM. As of Erlang 25, even Erlang JIT supports ARM64.
